filebeat/input/filestream/filestream_test.go

```diff
@@ -225,3 +227,91 @@ func readUntilError(reader *logFile) error {
 	}
 	return err
 }
+
+/*
+BenchmarkOffsetAndLastTimeRead was used the best approach between atomics
+and mutex to prevent race conditions when reading/writing to
+offset and lastTimeRead.
+
+Here are some results:
+goos: linux
+goarch: amd64
+pkg: github.com/elastic/beats/v7/filebeat/input/filestream
+cpu: 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
+BenchmarkOffsetAndLastTimeRead/atomic-16                1000000000               0.0000001 ns/op
+BenchmarkOffsetAndLastTimeRead/mutex-16                 1000000000               0.0000002 ns/op
+BenchmarkOffsetAndLastTimeRead/race-16                  1000000000               0.0000001 ns/op
+PASS
+ok      github.com/elastic/beats/v7/filebeat/input/filestream   9.850s
+
+% go test -bench=BenchmarkOffset -benchtime=60s
+goos: linux
+goarch: amd64
+pkg: github.com/elastic/beats/v7/filebeat/input/filestream
+cpu: 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
+BenchmarkOffsetAndLastTimeRead/atomic-16                1000000000               0.0000002 ns/op
+BenchmarkOffsetAndLastTimeRead/mutex-16                 1000000000               0.0000003 ns/op
+BenchmarkOffsetAndLastTimeRead/race-16                  1000000000               0.0000001 ns/op
+PASS
+ok      github.com/elastic/beats/v7/filebeat/input/filestream   9.723s
+
+% go test -bench=BenchmarkOffset -benchtime=30s
+goos: linux
+goarch: amd64
+pkg: github.com/elastic/beats/v7/filebeat/input/filestream
+cpu: 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
+BenchmarkOffsetAndLastTimeRead/atomic-16                1000000000               0.0000002 ns/op
+BenchmarkOffsetAndLastTimeRead/mutex-16                 1000000000               0.0000003 ns/op
+BenchmarkOffsetAndLastTimeRead/race-16                  1000000000               0.0000001 ns/op
+PASS
+ok      github.com/elastic/beats/v7/filebeat/input/filestream   9.728s
+*/
+func BenchmarkOffsetAndLastTimeRead(b *testing.B) {
+	a := benchAtomic{}
+	b.Run("atomic", func(b *testing.B) {
+		a.Inc()
+	})
+
+	m := benchMutex{}
+	b.Run("mutex", func(b *testing.B) {
+		m.Inc()
+	})
+
+	r := benchRace{}
+	b.Run("race", func(b *testing.B) {
+		r.Inc()
+	})
+}
```

Comment:
Seems to be missing the benchmark loop

```suggestion
func BenchmarkOffsetAndLastTimeRead(b *testing.B) {
	a := benchAtomic{}
	b.Run("atomic", func(b *testing.B) {
		for b.Loop() {
			a.Inc()
		}
	})

	m := benchMutex{}
	b.Run("mutex", func(b *testing.B) {
		for b.Loop() {
			m.Inc()
		}
	})

	r := benchRace{}
	b.Run("race", func(b *testing.B) {
		for b.Loop() {
			r.Inc()
		}
	})
}
```

Reply:
Thanks for spotting that!

I was gonna remove it from the PR because I don't believe it is relevant any more.

I've applied your fix and ran it, I'll add the results and code to the PR as a reference.